File: source/smc/generator/csharp/CSharpCodeGenerators/FSMAccessors.cs
namespace SMC.Generator.CSharp.CSharpCodeGenerators
{
    using System.Text;

    using SMC.Generator.CSharp;

    public class FSMAccessors : CSharpCodeGenerator
    {
        public override string GenerateCode(SMCSharpGenerator gen)
        {
            return new StringBuilder()
                .AppendLine("    #region Public Properties")
                .AppendLine()
                .AppendLine("    public string Version => version;")
                .AppendLine()
                .AppendLine("    public string CurrentStateName => this.currentState.Name;")
                .AppendLine()
                .AppendLine("    internal State CurrentState")
                .AppendLine("    {")
                .AppendLine("        get { return this.currentState; }")
                .AppendLine("        set { this.currentState = value; }")
                .AppendLine("    }")
                .AppendLine()
                .AppendLine("    #endregion")
                .AppendLine()
                .ToString();
        }
    }
}

File: source/smc/generator/csharp/CSharpCodeGenerators/FSMAccessors.java
package smc.generator.csharp.CSharpCodeGenerators;

import smc.generator.csharp.SMCSharpGenerator;

import java.util.List;

public class FSMAccessors extends CSharpCodeGenerator
{
    public String generateCode(SMCSharpGenerator gen)
    {
        StringBuffer buff = new StringBuffer();

        buff.append("    #region Public Properties\n");
        buff.append("\n");
        buff.append("    public string Version => version;\n");
        buff.append("\n");
        buff.append("    public string CurrentStateName => this.currentState.Name;\n");
        buff.append("\n");
        buff.append("    public State CurrentState\n");
        buff.append("    {\n");
        buff.append("        get { return this.currentState; }\n");
        buff.append("        set { this.currentState = value; }\n");
        buff.append("    }\n");
        buff.append("\n");
        buff.append("    #endregion\n");
        buff.append("\n");

        return buff.toString();
    }
}
